### ISSUE TITLE
How to get the word or text based on an element (i.e: link) when the range is 0 ? [Quill 0.20]

### ISSUE THREAD

**Issue by Vadorequest** [quilljs/quill#761]
Hi,

I have a link and I want to update it. If I select the whole link then I can use `@editor.getSelection()` to get the range and then apply it using `@editor.formatText(range, 'link', 'http://...', 'user')`.

But if I click on a link then my selection is 0 characters, because `range.start = range.end`. I'm looking for a way to apply the link change to the whole link, even when the user cursor is just in the link but doesn't select anything.

Long story short: If  `range.start = range.end` then select the whole link. Use the range otherwise.

Thanks!

**Comment by Vadorequest** [comment]
Solution found:

```

    _getActiveNodeRange: (range) ->
        [node, offset] = @editor.editor.doc.findLeafAt(range.start, true)
        start = range.start - offset
        end = start + node.length

        return start: start, end: end
```

is it possible to return a real range? This just return an object without method, while the usual `range` has methods like `isCollapsed`.

**Comment by jhchen** [comment]
You can create a new range in 0.20 with

```js
var Range = Quill.require('range');
var r = new Range(0, 100);
```

**Comment by Vadorequest** [comment]
Thanks!

Any hint about selecting a word when the cursor in inside that word but
nothing is selected? (range.start = range.end) I'm able to select a whole
node but I wonder if there is an easy way to get the active word itself.

I looked -quikly-at the source code but I didn't find anything like that.

2016-06-22 2:08 GMT+02:00 Jason Chen <notifications@github.com>:

> Closed #761 <https://github.com/quilljs/quill/issues/761>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/quilljs/quill/issues/761#event-699831243>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ADoY4ha-Qz652UH5v9UvcAG16pYHFRVAks5qOHz_gaJpZM4I6o3D>
> .
>



-- 
Cordialement,

M. Ambroise Dhenain.
